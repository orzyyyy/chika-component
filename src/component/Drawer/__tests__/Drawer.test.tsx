import React from 'react';
import { mount, shallow } from 'enzyme';
import 'nino-cli/scripts/setup';
let Drawer;
switch (process.env.LIB_DIR) {
  case 'lib':
    Drawer = require('../../../../lib/component/Drawer').default;
    break;
  default:
    Drawer = require('..').default;
    break;
}

describe('Drawer', () => {
  it('render correctly', () => {
    const wrapperShow = shallow(<Drawer visible={true}>test</Drawer>);
    const wrapperHide = shallow(<Drawer visible={false}>test</Drawer>);
    expect(wrapperShow).toMatchSnapshot();
    expect(wrapperHide).toMatchSnapshot();
  });

  it('when operate drawer is clicked, onChange should to be called', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Drawer visible={true} onChange={onChange} direction="right">
        test
      </Drawer>,
    );
    wrapper.find('.operate-drawer').simulate('click');
    expect(onChange).toBeCalled();
  });
});
