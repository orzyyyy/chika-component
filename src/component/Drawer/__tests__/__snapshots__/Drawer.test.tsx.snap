// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer render correctly 1`] = `
<Drawer
  visible={true}
>
  <div
    className="Drawer"
  >
    <div
      className="show-drawer"
      style={
        Object {
          "transform": "translateX(0px)",
          "width": 0,
        }
      }
    >
      test
    </div>
    <div
      className="operate-drawer"
      onClick={[Function]}
      style={
        Object {
          "background": "linear-gradient(to right, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0))",
          "transform": "translateX(0px)",
        }
      }
    >
      <div
        className="scale-wrapper"
        style={
          Object {
            "transform": "rotateZ(0deg)",
          }
        }
      >
        <img
          src="./assets/arrow.png"
        />
      </div>
    </div>
  </div>
</Drawer>
`;

exports[`Drawer render correctly 2`] = `
<Drawer
  visible={false}
>
  <div
    className="Drawer"
  >
    <div
      className="show-drawer"
      style={
        Object {
          "transform": "translateX(0px)",
          "width": 0,
        }
      }
    >
      test
    </div>
    <div
      className="operate-drawer"
      onClick={[Function]}
      style={
        Object {
          "background": "linear-gradient(to right, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0))",
          "transform": "translateX(0px)",
        }
      }
    >
      <div
        className="scale-wrapper"
        style={
          Object {
            "transform": "rotateZ(180deg)",
          }
        }
      >
        <img
          src="./assets/arrow.png"
        />
      </div>
    </div>
  </div>
</Drawer>
`;
